tests/test_quickfiles.py

```diff
@@ -0,0 +1,60 @@
+# -*- coding: utf-8 -*-
+import pytest
+
+from osf.models import QuickFiles, BaseFileNode
+from osf_tests.factories import AuthUserFactory
+
+from tests.base import test_app
+from webtest_plus import TestApp
+from addons.osfstorage.tests.utils import make_payload
+
+from framework.auth import signing
+
+
+@pytest.fixture()
+def user():
+    return AuthUserFactory()
+
+
+@pytest.fixture()
+def quickfiles(user):
+    return QuickFiles.objects.get(creator=user)
+
+
+@pytest.fixture()
+def flask_app():
+    return TestApp(test_app)
+
+
+@pytest.fixture()
+def post_to_quickfiles(quickfiles, user, flask_app, **kwargs):
+    def func(name, *args, **kwargs):
+        osfstorage = quickfiles.get_addon('osfstorage')
+        root = osfstorage.get_root()
+        url = '/api/v1/project/{}/osfstorage/{}/children/'.format(quickfiles._id, root._id)
+        expect_errors = kwargs.pop('expect_errors', False)
+        payload = make_payload(user=user, name=name, **kwargs)
+
+        res = flask_app.post_json(url, signing.sign_data(signing.default_signer, payload), expect_errors=expect_errors)
+        return res
+
+    return func
+
+
+@pytest.mark.django_db
+class TestUserQuickFilesFileCreation:
+
+    def test_create_file(self, quickfiles, user, post_to_quickfiles):
+        name = 'WoopThereItIs.pdf'
+
+        res = post_to_quickfiles(name)
+
+        assert res.status_code == 201
+        assert res.json['status'] == 'success'
+        assert BaseFileNode.objects.filter(type='osf.osfstoragefile', node__creator=user, name=name).exists()
+
+    def test_create_folder_throws_error(self, flask_app, user, quickfiles, post_to_quickfiles):
+        name = 'new_illegal_folder'
+        res = post_to_quickfiles(name, kind='folder', expect_errors=True)
+
+        assert res.status_code == 403
```

Comment:
I used a 403 instead of a 400 because it seemed to match the [RFC definitions]() -- 

400 Bad Request - The request could not be understood by the server due to malformed syntax.
vs
403 Forbidden
The server understood the request, but is refusing to fulfill it. Authorization will not help and the request SHOULD NOT be repeated.

You can't create a folder on quickfiles, and change of syntax won't help. If I'm misunderstanding or if we use 400 for consistency I'll definitely change it